spring-cloud-stream-binder-kinesis/src/main/java/org/springframework/cloud/stream/binder/kinesis/KinesisMessageChannelBinder.java

```diff
@@ -0,0 +1,117 @@
+import org.springframework.integration.aws.inbound.kinesis.KinesisMessageDrivenChannelAdapter;
+import org.springframework.integration.aws.inbound.kinesis.KinesisShardOffset;
+import org.springframework.integration.aws.inbound.kinesis.ListenerMode;
+import org.springframework.integration.aws.outbound.KinesisMessageHandler;
+import org.springframework.integration.channel.QueueChannel;
+
+/**
+ * 
+ * @author Peter Oates
+ */
+public class KinesisMessageChannelBinder extends
+		AbstractMessageChannelBinder<ExtendedConsumerProperties<KinesisConsumerProperties>, ExtendedProducerProperties<KinesisProducerProperties>, KinesisTopicProvisioner>
+		implements ExtendedPropertiesBinder<MessageChannel, KinesisConsumerProperties, KinesisProducerProperties> {
+
+	// not using currently but expect to include global properties here if
+	// required
+	private final KinesisBinderConfigurationProperties configurationProperties;
+
+	// not currently utilising any extended properties - not sure if I will need
+	// these
+	private KinesisExtendedBindingProperties extendedBindingProperties = new KinesisExtendedBindingProperties();
+
+	private AmazonKinesisAsync amazonKinesis;
+
+	public KinesisMessageChannelBinder(KinesisBinderConfigurationProperties configurationProperties,
+			KinesisTopicProvisioner provisioningProvider) {
+		super(false, null, provisioningProvider);
+		this.configurationProperties = configurationProperties;
+
+		this.amazonKinesis = AmazonKinesisAsyncClientBuilder.defaultClient();
+
+	}
+
+	@Override
+	public KinesisConsumerProperties getExtendedConsumerProperties(String channelName) {
+		return this.extendedBindingProperties.getExtendedConsumerProperties(channelName);
+	}
+
+	@Override
+	public KinesisProducerProperties getExtendedProducerProperties(String channelName) {
+		return this.extendedBindingProperties.getExtendedProducerProperties(channelName);
+	}
+
+	// below are the main methods to implement - these will create the message
+	// handlers used by the application
+	// to put and consume messages
+	@Override
+	protected MessageHandler createProducerMessageHandler(ProducerDestination destination,
+			ExtendedProducerProperties<KinesisProducerProperties> producerProperties) throws Exception {
+
+		KinesisMessageHandler kinesisMessageHandler = new KinesisMessageHandler(amazonKinesis);
+
+		kinesisMessageHandler.setStream(destination.getName());
+		// tidy up partition key
+		kinesisMessageHandler.setPartitionKey("name");
+		kinesisMessageHandler.setBeanFactory(this.getBeanFactory());
+
+		return kinesisMessageHandler;
+	}
+
+	@Override
+	protected MessageProducer createConsumerEndpoint(ConsumerDestination destination, String group,
+			ExtendedConsumerProperties<KinesisConsumerProperties> properties) throws Exception {
+
+		KinesisMessageDrivenChannelAdapter adapter = new KinesisMessageDrivenChannelAdapter(amazonKinesis,
+				destination.getName());
+
+		adapter.setOutputChannel(kinesisChannel());
```

Comment:
We don't need this. The `AbstractMessageChannelBinder` takes care about that for us:
```
consumerEndpoint = createConsumerEndpoint(destination, group, properties);
consumerEndpoint.setOutputChannel(bridge);
```